### DISCUSSION TITLE
Trail-Sense collaboration

### DISCUSSION THREAD

**Discussion by irwanwr** [kylecorry31/Trail-Sense#1326] [New Feature Ideas]
Is there a chance for this app developers to collaborate with developers of Trekarta and Dioptra™ Light? 
If there is, that would be awesome.

**Comment by kylecorry31** [comment]
It doesn't look like Dioptra is under active development (Trekarta is though)

But as with other map apps, I would most likely limit my integration to the standard system support for opening locations and opening GPX files - both of which Trail Sense already supports and would require the other apps to support them as well. I would also put a link as a recommended app once I test them out and review their licenses / code

Is there another type of integration you have in mind?

**Reply by irwanwr** [reply]
i was thinking if it would also have features that works like sextant. or, any other astronomical geonav device.

**Reply by irwanwr** [reply]
also rangefinder or similar, to estimate distance or height. based on phone (observer eyes) height, and view vertical angle, using auto calculation of trigonometry. thus why i mentioned dioptra earlier. 
with dioptra, one can put the phone on stick/pole/tripod leveled, ensure the height, then turn the view vertically to point the object's base/bottom. put the cam height and angle into calculation to determine est distance.

**Reply by kylecorry31** [reply]
Hi, I believe the features you are asking for are already present in the Clinometer tool within Trail Sense (the ruler button on the top right of the tool will allow you to estimate the distance or height of an object). Based on my research to estimate the height of an object you need to know the distance away from it and its angular height. Alternatively, to measure the distance to an object, you need to know the height of the object and its angular height. I could not find a way to do that using just information provided by the phone (ex. angular size only) - but if you know of a way, I would be happy to implement that.

Unfortunately, a sextant requires more accuracy than a phone is capable of providing - but the clinometer can be used for a rough angle estimate (research on this was conducted in #319)

**Reply by irwanwr** [reply]
i am sorry. didn't know that it's featured on clinometer. thus i mentioned dioptra. 
it's about the camera or the observer relative height, to the base line as the base height to calculate. 
crossed my mind and after seeing this https://stonekick.com/blog/using-basic-trigonometry-to-measure-distance.html#:~:text=B%20(distance)%20%3D%20A%20(,calculating%20using%20degrees%20not%20radians). 
i guess you've figured that out too.

**Reply by kylecorry31** [reply]
No worries - yes, that link contains a very similar formula to the one I'm using to get distance (mine is a little more complicated because it doesn't assume the ground level is constant - ex. measuring the height of a tree downhill from you)
